key_management/src/wallet.rs

```diff
@@ -72,7 +72,14 @@ where
             return Ok(k.clone());
         }
         let key_string = format!("wallet-{}", addr.to_string());
-        let key_info = self.keystore.get(&key_string)?;
+        let key_info = match self.keystore.get(&key_string) {
```

Comment:
can use [map_err](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err) here

Reply:
Nah, map_err transforms the error type. I think you mean't  `unwrap_or_else`, but that doesnt work either because the closure is supposed to return the same type, whereas im calling another method that returns a Result

Reply:
Maybe you can use [or_else](https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else) here?